### DISCUSSION TITLE
200 -> 403 for unauthenticated API requests

### DISCUSSION THREAD

**Discussion by ligi** [pretalx/pretalx#1496] [General]
I see that unauthenticated API requests return an code 200 - but empty objects like this:
```json
{"count":0,"next":null,"previous":null,"results":[]}
```

It feels a bit weird that it does not return a 403 in this case. Is this a design decision that I fail to understand? Currently I think a 403 would be better in this case.

**Comment by rixx** [comment]
Some, if not most, API endpoints can return data to unauthenticated users too. For example, the list of proposals will only return published (accepted, confirmed, scheduled) proposals to unauthenticated users. If there are none, the API will indicate this with an empty list, as access to the endpoint is not prohibited to unauthenticated users.

**Comment by ligi** [comment]
I think it would still be better to return a 403 here - otherwise you do not know if there was a problem with auth or if the list is just empty. It is OK that other endpoints return a 200 for unauthenticated users when they return data - but in this case IMHO 403 is correct.

**Reply by q9f** [reply]
IMO returning 0 results if nothing was published should be the correct behaviour. You usually don't use this API authed - for that you can use the access token.

**Reply by ligi** [reply]
maybe a field then signaling it is unauthorized? This behavior already resulted in some confusion - hence the request.

**Reply by rixx** [reply]
The pretalx API is generally designed to permit both authenticated and unauthenticated requests. Quite a bit of data is public, for example speaker profiles, schedule data, etc., so a 403 is entirely unpractical.
You can always use the `/api/me` endpoint to check who you are logged in as ([cf docs](https://docs.pretalx.org/api/fundamentals.html#obtaining-an-api-token)).
